src/core/network/qgsfiledownloader.h

```diff
@@ -47,6 +47,17 @@ class CORE_EXPORT QgsFileDownloader : public QObject
     Q_OBJECT
   public:
 
+    /**
+     * Different methods of HTTP requests supported by QgsFileDownloader
+     * \since 3.22
+     */
+    enum HttpMethod
```

Comment:
Can you move this enum to the Qgis class? (And follow the pattern used there with enum class, Q_ENUM etc)